### PR TITLE
llm-load-test model dependent dataset sample size

### PR DESCRIPTION
**Pull request by npalaska**

### CODE REVIEW

**Comment by npalaska** [comment]
`watsonx_single_model_gating_light` [test passed](https://ci.app-svc-perf.corp.redhat.com/job/ExternalTeams/job/RHODS/job/topsail/173/) on dgx machine

**Comment by npalaska** [comment]
[Test succeeded](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/273/pull-ci-openshift-psap-topsail-main-rhoai-light/1766974698177957888/artifacts/light/test/artifacts/) on `kserve quick` (caikit+tgis)

**Comment by npalaska** [comment]
`watsonx_single_model_gating_large` [Text passed](https://ci.app-svc-perf.corp.redhat.com/job/ExternalTeams/job/RHODS/job/topsail/174/) on dgx machine for the larger models

File: projects/kserve/testing/test_e2e.py [hunk 1]
```diff
@@ -620,6 +620,14 @@ def test_consolidated_model(consolidated_model, namespace=None):
...
     llm_load_test_args = config.ci_artifacts.get_config("tests.e2e.llm_load_test.args")
 
+    size_name = consolidated_model.get("testing")
+    if not size_name:
+        size_name = "small"
+    else:
+        size_name = size_name.get("size", "small")
+    llm_load_test_dataset_sample_args = config.ci_artifacts.get_config(f"tests.e2e.llm_load_test.dataset_size.{size_name}")
+    llm_load_test_args |= llm_load_test_dataset_sample_args
+
```
Comment (kpouget) [inline projects/kserve/testing/test_e2e.py:630]: Thanks @npalaska , looks nearly all good :)

can you change this:
```suggestion
    size_name = consolidated_model["testing"]["size"]

    llm_load_test_dataset_sample_args = config.ci_artifacts.get_config(f"tests.e2e.llm_load_test.dataset_size.{size_name}")
    llm_load_test_args |= llm_load_test_dataset_sample_args
```
and include this in `base.yaml` and `base-standalone-tgis.yaml`:
```
testing:
  size: small
```
(and you can remove the `testing.size.small` in the other files)

this removes the hard-coded default `small` property in the code

and just quick remark:
```
size_name = consolidated_model.get("testing")
size_name = size_name.get("size", "small")
```
this naming was not intuitive, should have been something like this, to make it more readable:
```
testing_props = consolidated_model.get("testing")
size_name = testing_props.get("size", "small")
```

**Comment by kpouget** [comment]
> [Test succeeded](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/273/pull-ci-openshift-psap-topsail-main-rhoai-light/1766974698177957888/artifacts/light/test/artifacts/) on `kserve quick` (caikit+tgis)

thanks for the test links, it's great this way 🎉

File: projects/kserve/testing/test_e2e.py (hunk 1 above)
Comment (npalaska) [reply projects/kserve/testing/test_e2e.py]: Fixed.

**Comment by kpouget** [comment]
[test passed](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-psap_topsail/273/pull-ci-openshift-psap-topsail-main-rhoai-light/1767456834257424384/artifacts/light/test/artifacts/report_index.html), merging

thanks @npalaska for the work :)

**Pull request merged**